src/libraries/System.Formats.Tar/tests/System.Formats.Tar.Tests.csproj

```diff
@@ -101,6 +101,7 @@
     <Compile Include="$(CommonPath)Interop\Unix\System.Native\Interop.Stat.cs" Link="Common\Interop\Unix\System.Native\Interop.Stat.cs" />
   </ItemGroup>
   <ItemGroup>
+    <ProjectReference Include="..\src\System.Formats.Tar.csproj" />
```

Comment:
@ViktorHofer, is this desirable?

Reply:
@carlossanlop the list above include out-of-band projects only which require the ProjectReference as otherwise they would be missing the compiler reference.

System.Formats.Tar is part of the targeting pack and hence doesn't need to be referenced by the test project.

> @ViktorHofer, is this desirable?

I guess what Carlos is trying to achieve with this change is that a test project's build also keeps the underlying source project up-to-date by building it. That's the behavior that exists for OOBs already today but for inbox libraries we weren't able to find a good pattern that supports that. It works in some cases like this one but not in others when the test projects is platform agnostic but the source project is platform specific.

The not so obvious downside of this change is that the build graph with the Formats.Tar.Tests project as the entry point gets bigger because it now also references Formats.Tar and all its dependencies transitively.

Reply:
Thanks for the explanation. What is your preference then? Should I revert this to avoid the downside you mentioned?

Reply:
I would prefer consistency in this case by not having the ProjectReference but I see that it is desirable to have. Two years ago I was working on a PR that would add ProjectReferences to the src to all the test projects and adapt the infrastructure to fully support that. Unfortunately I was never able to finish that work.

Just found that branch: https://github.com/dotnet/runtime/compare/main...ViktorHofer:runtime:RefSrcFromTest.

Reply:
> the build graph with the Formats.Tar.Tests project as the entry point gets bigger
> I would prefer consistency in this case by not having the ProjectReference

I guess we can solve this by rehydrating your effort from that old commit. I prefer to not cause the problem you described, so I'll revert the csproj change.

Reply:
Not suggesting it, but another option presumably is to add it, but with a condition defaulting to false. Then if Carlos wanted this behavior locally he could just Eg define a environment variable to switch it on for himself.